### ISSUE TITLE
Running Django example app return UnrecognizedInputFormatException

### ISSUE THREAD

**Issue by maclogan** [gunthercox/ChatterBot#359]
Running the Django example app listed under examples with no modificiation returned "chatterbot.adapters.input.variable_input_type_adapter.VariableInputTypeAdapter.UnrecognizedInputFormatException: "This type <'NoneType'> is not recognized as a valid input type."

I feel like this is an error on my end but I can't seem to resolve it.

**Comment by vkosuri** [comment]
I think you have to add chatbot init to settings.py. I am using like this

I am getting authentication issue, so I have commented CsrfViewMiddleware

``` Python
MIDDLEWARE_CLASSES = (
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    # 'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'django.middleware.security.SecurityMiddleware',
)
```
``` Python
CHATTERBOT = {
    'name': 'ABOT',
    'logic_adapters': [
        'chatterbot.adapters.logic.TimeLogicAdapter',
        chatterbot.adapters.logic.MathematicalEvaluation',     
    ]
}
```

**Comment by maclogan** [comment]
Thanks for the response, but this didn't seem to work for me.  I am still getting the same error message as before.

**Comment by vkosuri** [comment]
@maclogan could you paste you console output, By the way a quote was missed

``` Python
CHATTERBOT = {
    'name': 'Test',
    'logic_adapters': [
        'chatterbot.adapters.logic.TimeLogicAdapter',
        'chatterbot.adapters.logic.MathematicalEvaluation',     
    ]
}
```

More info available here http://chatterbot.readthedocs.io/en/stable/django/settings.html

**Comment by maclogan** [comment]
This is the full error I get after hitting the submit button on the online UI provided in the django app example.


`Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.5/dist-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/usr/local/lib/python3.5/dist-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.5/dist-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.5/dist-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/chatterbot/ext/django_chatterbot/views.py", line 14, in post
    response_data = self.chatterbot.get_response(input_statement)
  File "/usr/local/lib/python3.5/dist-packages/chatterbot/chatterbot.py", line 138, in get_response
    input_statement = self.input.process_input(input_item)
  File "/usr/local/lib/python3.5/dist-packages/chatterbot/adapters/input/variable_input_type_adapter.py", line 39, in process_input
    input_type = self.detect_type(statement)
  File "/usr/local/lib/python3.5/dist-packages/chatterbot/adapters/input/variable_input_type_adapter.py", line 34, in detect_type
    input_type
chatterbot.adapters.input.variable_input_type_adapter.VariableInputTypeAdapter.UnrecognizedInputFormatException: "The type <class 'NoneType'> is not recognized as a valid input type."
`

**Comment by gunthercox** [comment]
@maclogan What version of Django do you have installed?
You can run `python -m django --version` to check this.

**Comment by maclogan** [comment]
I have 1.10.2 installed

**Comment by gunthercox** [comment]
Hmm, OK. I can't create this issue testing with Python 3, Django 1.10.2 and the latest version of ChatterBot.

Is there any chance that you are using the old example code from https://github.com/gunthercox/django_chatterbot (which is outdated and no longer maintained)

instead of https://github.com/gunthercox/ChatterBot/tree/master/examples/django_app (which is the latest, tested copy of the example)?

**Comment by maclogan** [comment]
Hm I must be doing something wrong then.  I am using the current example app and have just tried it on another python/django installation and it didn't work

**Comment by maclogan** [comment]
So just to be sure I am not missing anything obvious, to run the example app all I have to do is be sure my django, chatterbot, and example/django-app are up to date, modify the setting.py file to add logic adapters to the chatterbot settings, and run a migration.  Then I just runserver like usual, correct?

**Comment by vkosuri** [comment]
yes

**Comment by gunthercox** [comment]
@maclogan What version of ChatterBot do you have installed?

**Comment by maclogan** [comment]
chatterbot-0.4.10

**Comment by gunthercox** [comment]
Ok, I was just able to replicate the issue you are having. I've released ChatterBot `0.4.11` which appears to fix this problem. What to upgrade and check if the issue is fixed for you? You should be able to run `pip install chatterbot --upgrade` to get the latest copy.

**Comment by maclogan** [comment]
That seems to have fixed it!  Sorry for the trouble, I ran an update on chatterbot 2 days ago, but I just checked my logs and I guess it failed during the installation.
